torch/distributed/_tensor/_utils.py

```diff
@@ -149,6 +149,8 @@ def compute_global_tensor_info(
                 shard_placement.dim += len(tensor_shape)
             shard_dim = shard_placement.dim
 
+            assert shard_dim < tensor.ndim, f"Sharding dim {shard_dim} greater than tensor ndim {tensor.ndim}"
```

Comment:
It'd be good if this error message explained how this issue relates to `placements` and suggests what the user should do to resolve it.